discovery/azure/azure.go

```diff
@@ -597,35 +627,19 @@ var errorNotFound = errors.New("network interface does not exist")
 // getNetworkInterfaceByID gets the network interface.
 // If a 404 is returned from the Azure API, `errorNotFound` is returned.
 // On all other errors, an autorest.DetailedError is returned.
-func (client *azureClient) getNetworkInterfaceByID(ctx context.Context, networkInterfaceID string) (*network.Interface, error) {
-	result := network.Interface{}
-	queryParameters := map[string]interface{}{
-		"api-version": "2018-10-01",
-	}
-
-	preparer := autorest.CreatePreparer(
-		autorest.AsGet(),
-		autorest.WithBaseURL(client.nic.BaseURI),
-		autorest.WithPath(networkInterfaceID),
-		autorest.WithQueryParameters(queryParameters),
-		autorest.WithUserAgent(userAgent))
-	req, err := preparer.Prepare((&http.Request{}).WithContext(ctx))
-	if err != nil {
-		return nil, autorest.NewErrorWithError(err, "network.InterfacesClient", "Get", nil, "Failure preparing request")
-	}
-
-	resp, err := client.nic.GetSender(req)
-	if err != nil {
-		return nil, autorest.NewErrorWithError(err, "network.InterfacesClient", "Get", resp, "Failure sending request")
-	}
+func (client *azureClient) getNetworkInterfaceByID(ctx context.Context, networkInterfaceID string) (*armnetwork.Interface, error) {
+	splitedNetworkInterfaceID := strings.Split(networkInterfaceID, "/")
+	resourceGroupName := splitedNetworkInterfaceID[4]
+	networkInterfaceName := splitedNetworkInterfaceID[8]
 
-	result, err = client.nic.GetResponder(resp)
+	resp, err := client.nic.Get(ctx, resourceGroupName, networkInterfaceName, nil)
```

Comment:
I Don't like this implementation but I couldn't find a way to use the ID directly. I guess Azure wants to keep the subscriptionID an internal data in the NewInterfacesClient

Reply:
I think you could use `ParseResouceID` function in `azcore.arm` to do such things. Please refer: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azcore@v1.3.0/arm#ParseResourceID.

Reply:
Thank you ! Integrated in the existing `newAzureResourceFromID` method